.travis.yml

```diff
@@ -1,32 +1,64 @@
-language: rust
-dist: trusty
-sudo: required
+language: python
+sudo: false
+
+python: "3.4"
 
 env:
+  global:
     - PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$HOME/install/syslog-ng/lib/pkgconfig
+    - CMAKE_C_FLAGS=-std=gnu99
+    - LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$HOME/install/syslog-ng/lib"
+    - PATH="$HOME/.local/bin:$PATH"
+  matrix:
+    - RUST_CHANNEL=stable TEST_MODE=everything
+    - RUST_CHANNEL=beta TEST_MODE=everything
+    - RUST_CHANNEL=nightly TEST_MODE=everything TRAVIS_CARGO_NIGHTLY_FEATURE=nightly
```

Comment:
You should probably list the slowest ones at the top. In my experience, Travis queues from top to bottom.